spec/lib/traject/macros/custom_spec.rb

```diff
@@ -0,0 +1,134 @@
+
+    it 'produces a fulltext link' do
+      expect(result_1['full_links_struct']).to match ['{"text":"scholarsphere.psu.edu","url":"https://scholarsphere.ps'\
+                                                      'u.edu/files/02870v85d"}']
+    end
+  end
+
+  describe 'A record without a magic word in one of the label subfields' do
+    let(:url_856_2) do
+      { '856' => { 'ind1' => '0', 'ind2' => '7', 'subfields' => [{ 'u' => 'https://scholarsphere.psu.edu/files/02870v8'\
+                                                                           '5d' },
+                                                                 { '3' => 'Apples bananas' }] } }
+    end
+    let(:result_2) { @indexer.map_record(MARC::Record.new_from_hash('fields' => [url_856_2], 'leader' => leader)) }
+
+    it 'produces a fulltext link' do
+      expect(result_2['full_links_struct']).to eq ['{"text":"scholarsphere.psu.edu","url":"https://scholarsphere.psu.e'\
+                                                                                          'du/files/02870v85d"}']
+    end
+  end
+
+  describe 'A record with an indicator 2 of 2 and magic word is in one of the label subfields' do
+    let(:url_856_3) do
+      { '856' => { 'ind1' => '0', 'ind2' => '2', 'subfields' => [{ 'u' => 'http://library.columbia.edu/content/library'\
+                                                                   'web/indiv/ccoh/our_work/how_to_use_the_archives.ht'\
+                                                                   'ml' },
+                                                                 { '3' => 'Carrots executive summary peas' }] } }
+    end
+    let(:result_3) { @indexer.map_record(MARC::Record.new_from_hash('fields' => [url_856_3], 'leader' => leader)) }
+
+    it 'produces a supplemental link' do
+      expect(result_3['suppl_links_struct']).to match ['{"text":"library.columbia.edu","url":"http://library.columbia.'\
+                                                       'edu/content/libraryweb/indiv/ccoh/our_work/how_to_use_the_arch'\
+                                                       'ives.html"}']
+    end
+  end
+
+  describe 'A record with an indicator 2 of 2 and magic word is not one of the label subfields' do
+    let(:url_856_3a) do
+      { '856' => { 'ind1' => '0', 'ind2' => '2', 'subfields' => [{ 'u' => 'http://library.columbia.edu/content/library'\
+                                                                   'web/indiv/ccoh/our_work/how_to_use_the_archives.ht'\
+                                                                   'ml' },
+                                                                 { '3' => 'Carrots peas' }] } }
+    end
+    let(:result_3a) { @indexer.map_record(MARC::Record.new_from_hash('fields' => [url_856_3a], 'leader' => leader)) }
+
+    it 'produces a supplemental link' do
+      expect(result_3a['full_links_struct']).to match ['{"text":"library.columbia.edu","url":"http://library.columbia.'\
```

Comment:
@cdmo in this case, does it produce a supplemental or a fulltext link? or maybe just the description needs to be changed here?

Reply:
full. oops.